policies/coding-style.md

```diff
@@ -323,7 +323,17 @@ because it is a simple way to add valuable information for the reader.
 The name in the prototype declaration should match the name in the function
 definition.
 
-### Chapter 6.1: Extending existing functions
+### Chapter 6.1: Checking function aguments
+
+A _public_ function should verify that it's arguments are sensible.
+This includes, but is not limited to, verifying that:
+- non-optional pointer arguments are not NULL;
+- enumerated arguments have a meaningful value and
```

Comment:
I think this should also cover things like flags using bits, where it should reject unknown flags.

Reply:
I'll add this specifically.  It could be assumed but best not.

Reply:
I am worried that we are taking this argument checking too far.

Reply:
Let's say that there is a new flag introduced to for example enable some security hardening in later releases. The older releases could work fine as well without enabling this hardening but since the flag passed in will be unknown the function call will fail.

Reply:
I agree with @t8m. Adding such checks could be unnecessarily constraining

Reply:
I would even add that the *_ctrl() function should check that command is a known command, and the arguments (if any) might also have a limited amount of valid values. So I would in general say that if an argument should be check to be a known valid option.

Not checking that they are valid values almost always leads to problems. As far as I know, people have problems with 3.0 because we did not add such checks. Ignoring the unknown value can lead to security problems because they think your security hardening is available and have no way to check that it's not available and fall back to an other way to do it.

Reply:
I've reverted this change.  I think @t8m hit a use case we want to work.

Reply:
Most of the ctrl functions do check that the command is known.
I agree, they should also check for valid arguments.  As should params.

Reply:
I really don't understand the argument. If a function doesn't exist in an older version, we're happy with a (linker) error. If one of the control commands doesn't exist, we're happy with an error. But for the special case of flags, returning an error on an unknown flag is somehow problematic. Those things really are just the same thing, things we don't known about. Pretending we do known it and did and process the request is causing people problems.